posts/auditing-using-javers.md

```diff
@@ -0,0 +1,240 @@
+- JaVers is a lightweight Java library for auditing changes in your data. It can be used with any kind of database and any kind of persistence framework since the audit data is easily decoupled from live data; we can choose where to store it.
+
+
+# Audit trail with JaVers
+Since our client is mainly using MongoDB, we chose to use JaVers.
+JaVers is an audit log framework that helps to track changes of entities in the application.
+
+The usage of this tool is not limited to debugging and auditing only. It can be successfully applied to perform analysis and force security policies.
+
+# Spring Boot Integration and Setup
+JaVers provides a Spring Boot starter that integrates all the required JaVers beans with the default configuration.
+
+pom.xml
+```xml
+<dependency>
+            <groupId>org.javers</groupId>
+            <artifactId>javers-spring-boot-starter-mongo</artifactId>
+            <version>5.8.8</version>
+</dependency>
+```
+
+
+# JaVers repository configuration
+JaVers [Spring Boot starter](https://javers.org/documentation/spring-boot-integration/) is configured to reuse the application's persistent data store.  This is managed by Spring Data.
+Alternatively, we can choose to store JaVers data in a separate database, as shown below:
+
+```yaml
+javers:
+  mongodb:
+    host: localhost
+    port: 27017
+    database: javers-audit
+    authentication-database: admin
+    username: javers
+    password: password
+```
+Or
+
+```yaml
+javers:
+  mongodb:
+    uri: mongodb://javers:password@localhost:27017/javers-audit&authSource=admin
+```
+
+# Auto audit annotations
+
+Auditing specific entities can be done quickly without additional code. Simply add the `@JaversSpringDataAuditable` annotation on the corresponding Spring Data repositories.
+
+Example on MongoRepository
+```
+import org.javers.spring.annotation.JaversSpringDataAuditable
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.javers.organization.structure.domain.Person;
+
+@JaversSpringDataAuditable
+public interface PersonRepository extends MongoRepository<Person, String> {
+}
+```
+JaVers will audit all the changes made to Person objects.
+
+In case of a custom repository, the `@JaversAuditable` annotation needs to be added at the method level:
+
+```java
+@Repository
+class UserRepository {
+
+    @JaversAuditable
+    public void save(User user) {
+        ...//
+    }
+
+    public User find(String login) {
+        ...//
+    }
+}
+```
+
+For fine-grained control, the JaVers Spring bean can be used:
+
+```java
+@Service
+class PersonService {
+    private final Javers javers;
+    private final PersonRepository personRepository;
+
+    @Autowired
+    public PersonService(Javers javers, PersonRepository personRepository) {
+        this.javers = javers;
+        this.personRepository = personRepository;
+    }
+
+    public void fire(Person person) {
+        person.fire();
```

Comment:
I'd suggest finding another example than firing people on a Friday haha

Reply:
Plus, for a second, a thought `fire` was actually part of the Javers api (like: "fire and audit event")

Reply:
Done